applications/archive/archive.c

```diff
@@ -144,26 +130,199 @@ static void set_file_type(ArchiveFile_t* file, FileInfo* file_info) {
     }
 }
 
-static bool archive_get_filenames(ArchiveApp* archive) {
+static void archive_file_append(ArchiveApp* archive, const char* path, string_t string) {
+    furi_assert(archive);
+    furi_assert(path);
+    furi_assert(string);
+
+    File* file = storage_file_alloc(archive->api);
+    bool result = storage_file_open(file, path, FSAM_WRITE, FSOM_OPEN_APPEND);
+
+    if(result) {
+        uint16_t bytes_count =
+            storage_file_write(file, string_get_cstr(string), string_size(string));
+        if(bytes_count != string_size(string)) {
+            result = false;
+        }
+    } else {
+        FURI_LOG_E("Archive", "Cant open file");
+    }
+
+    result = storage_file_close(file);
+
+    if(!result) {
+        FURI_LOG_E("Archive", "File close, error");
+    }
+
+    storage_file_free(file);
+}
+
+static void archive_view_add_item(ArchiveApp* archive, FileInfo* file_info, const char* name) {
     furi_assert(archive);
+    furi_assert(file_info);
+    furi_assert(name);
 
     ArchiveFile_t item;
+
+    if(filter_by_extension(archive, file_info, name)) {
+        ArchiveFile_t_init(&item);
+        string_init_set_str(item.name, name);
+        set_file_type(&item, file_info);
+
+        with_view_model(
+            archive->view_archive_main, (ArchiveViewModel * model) {
+                files_array_push_back(model->files, item);
+                return true;
+            });
+
+        ArchiveFile_t_clear(&item);
+    }
+}
+
+static bool archive_favorites_action(
```

Comment:
Writing a "superfunction" is bad practice, try to split this into separate functions for each action. 

We also have file_worker in the library, which makes working with files much easier, for example, it has the read_until method that you invented here.